### PR TITLE
DOCS-2313: Clean up gantry, camera, board, encoder and arm pages for docs parsing

### PR DESCRIPTION
**Pull request by sguequierre**
Questions: 
* Is there any way to manually add a link for a type in the docstrings?

### CODE REVIEW

**Comment by andf-viam** [comment]
Great start! Thanks @sguequierre !

- Yes, you should omit `timeout` and `extra`. We scrape these from the initial usage declaration of the method (first line) and not from the **Properties** section. I believe the intend is to omit these SDK-wide. Only `motion` included them explicitly, and should likely remove them also.
- Yes, for docs.viam.com. We can manually add a link using the `python_datatype_links` array in `update_sdk_methods.py`. For the upstream SDK docs site though, I'd ask the team here: "any way to manually provide links here outside of the automated linking that occurs during `make documentation`?" Sierra, you made an excellent observation that some existing generated links by this step appear incorrect: "the resource type links are a bit tricky, as far as I can tell this is done automatically by some scraper/parser and it is not always correct (See base [Properties](https://python.viam.dev/autoapi/viam/components/base/client/index.html#viam.components.base.client.BaseClient.get_properties) linking to audioinput).

**Review by stuqdog — CHANGES_REQUESTED** [review]
Nice, thanks for doing this! A couple small things, but otherwise this looks great.

File: src/viam/components/arm/arm.py [hunk 1]
```diff
@@ -44,8 +44,14 @@ async def get_end_position(
             # Get the end position of the arm as a Pose.
             pos = await my_arm.get_end_position()
 
+        Args:
+            extra (Optional[Dict[str, Any]]): Extra options to pass to the underlying RPC call.
+            timeout (Optional[float]): An option to set how long to wait (in seconds) before calling a time-out and closing the underlying RPC call.
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply src/viam/components/arm/arm.py:49]: Ok, thank you!

**Comment by stuqdog** [comment]
> > * any way to manually provide links here outside of the automated linking that occurs during `make documentation`?
> 
> @stuqdog do you know about this? No worries if not

Ah good question! I'm not sure, but @njooma might know?

**Review by andf-viam — APPROVED** [review]
LGTM, thank you!

**Pull request merged**